docs/source/tensor_attributes.rst

```diff
@@ -210,3 +210,25 @@ Example::
     (1, 5)
 
 For more information on ``torch.sparse_coo`` tensors, see :ref:`sparse-docs`.
+
+torch.memory_format
+------------
+
+.. class:: torch.memory_format
+
+A :class:`torch.memory_format` is an object representing the memory format on which a :class:`torch.Tensor` is
```

Comment:
it's a bit non-idiomatic to structure things like "X is an object representing the X", it's more natural to say something like "X is an object representing 'longer description of X'".  Also remove the "A" at the beginning of the sentence.

Basically I'd expect some description about physical memory layout vs semantic dimension meaning here.

Something like (please improve!):

> `torch.memory_format` is an object that represents how the dimensions of a Tensor are physically laid out in memory.  Memory format allows users to separately control the physical ordering and semantic meaning of the dimensions of a Tensor.  For example, a user may wish to semantically represent a batch of images as having dimension order [N, C, H, W] (to match PyTorch's convention), but may wish to order the dimensions in memory as [N, H, W C] in order to take advantage of (possibly third-party) optimized algorithms for Tensors in that physical layout.